### PR TITLE
feat: personalized billing cycles anchored to user subscription/creation date

### PR DESCRIPTION
**Pull request by AchoArnold**
## Summary

- Replace calendar-month billing windows with personalized cycles anchored to each user's subscription renewal date (paid) or account creation date (free)
- Billing cycle boundaries are computed dynamically at runtime — no new database columns needed
- Usage history now displays full date ranges (e.g. "May 15, 2026 – June 14, 2026") and removes the Total Cost column

### API
- **pi/pkg/entities/billing_cycle.go** — New ComputeBillingCycle() pure function with dynamic day-of-month clamping for short months
- **pi/pkg/entities/user.go** — New GetBillingAnchorDay() method (derives anchor from SubscriptionRenewsAt or CreatedAt)
- **pi/pkg/repositories/gorm_billing_usage_repository.go** — Range-based queries (start_timestamp <= ? AND end_timestamp >= ?) replace exact calendar-month matching; user fetch only on new cycle creation
- **pi/pkg/di/container.go** — Inject UserRepository into billing usage repository

### Web
- **web/plugins/filters.ts** — New illingPeriodDate filter; updated illingPeriod filter to show date ranges
- **web/pages/billing/index.vue** — Usage history shows full billing period; Total Cost column removed

## Test Plan

- [x] 14 unit tests for ComputeBillingCycle (edge cases: Feb, leap year, day 31, year boundary)
- [x] 5 unit tests for GetBillingAnchorDay (free/paid/nil scenarios)
- [x] Full API build passes
- [x] All repository and service tests pass

### CODE REVIEW

File: web/plugins/filters.ts [hunk 1]
```diff
@@ -45,12 +45,32 @@ Vue.filter('decimal', (value: string): string => {
...
+    day: 'numeric',
     year: 'numeric',
+  }
+  const start = startDate.toLocaleDateString('en-US', options)
+  const endDate = new Date(startDate)
+  endDate.setMonth(endDate.getMonth() + 1)
+  endDate.setDate(endDate.getDate() - 1)
+  const end = endDate.toLocaleDateString('en-US', optionsWithYear)
+  return `${start} – ${end}`
+})
```
[the remark this answers is not in the document]
Comment (AchoArnold) [reply web/plugins/filters.ts:55]: Fixed in d17aba70. The billingPeriod filter now simply formats a single date. The Overview section uses billingPeriodDate applied to both start_timestamp and end_timestamp directly from the store, matching the server-computed cycle exactly. Also removed the UserRepository dependency - user is now loaded directly within the CockroachDB transaction via tx to ensure consistent snapshot reads on retries.

**Comment by AchoArnold** [comment]
Both issues raised by Greptile are fixed in d17aba70:

**P1 (Overview date range diverges from server):** The Overview section now uses \illingPeriodDate\ applied to both \start_timestamp\ and \nd_timestamp\ directly from the store, eliminating the JS arithmetic that diverged from server clamping.

**P2 (User load outside transaction):** \createBillingUsageForUser\ now accepts the \	x *gorm.DB\ parameter and loads the user directly via the transaction, ensuring consistent snapshot reads across CockroachDB retries. The \UserRepository\ dependency has been removed from \BillingUsageRepository\ entirely.
